icu4c/source/i18n/reldatefmt.cpp

```diff
@@ -605,6 +605,9 @@ static void loadWeekdayNames(UnicodeString absoluteUnits[UDAT_STYLE_COUNT]
                                  [UDAT_ABSOLUTE_UNIT_COUNT][UDAT_DIRECTION_COUNT],
                              const char* localeId,
                              UErrorCode& status) {
+    if (U_FAILURE(status)) {
+        return;
+    }
     Locale locale(localeId);
     DateFormatSymbols dfSym(locale, status);
```

Comment:
Since the DateFormatSymbols constructor internally allocates memory (in various places), could we also check the `status` again after the line: `DateFormatSymbols dfSym(locale, status);` ?
(Otherwise, if OOM occurs, the dfSym object will be half-constructed and the code might still crash in the loop below).

Reply:
Good point. I'll add that.

Reply:
Thanks!